### ISSUE TITLE
The "../.." folder is always ignored

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#10675]
(Found while investigating [this report](https://gitter.im/eslint/eslint?at=5b594d8a1be9bb57bcc4222d) from Gitter)

**Tell us about your environment**

* **ESLint Version:** v5.2.0
* **Node Version:** 8.11.3
* **npm Version:** 5.10.0

**What parser (default, Babel-ESLint, etc.) are you using?**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
(none)
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```bash
mkdir -p foo/bar/baz
echo 'foo!' > foo/file.js
cd foo/bar/baz
eslint --no-eslintrc ../..
```

**What did you expect to happen?**

I expected ESLint to lint the `foo/file.js` file and report a parsing error for it, since its syntax is invalid.

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint didn't find any files:

```
$ eslint --no-eslintrc ../..

Oops! Something went wrong! :(

ESLint: 5.2.0.
No files matching the pattern "../.." were found.
Please check for typing mistakes in the pattern.
```

This is happening as a result of the logic [here](https://github.com/eslint/eslint/blob/acb6658081eeed7e114d69b1ccf2c22c34da69c9/lib/ignored-paths.js#L270). By default, ESLint ignores all folders starting with a `.` character (to prevent things like `.git` from being linted), and there is supposed to be an exception that prevents folders in `..` from being ignored. However, it seems like this exception isn't working when a folder is `../..`, so that folder is incorrectly being ignored.

**Comment by gssridhar12** [comment]
@not-an-aardvark Is this taken? I want to fix this.
